src/commands/dev/socket.rs

```diff
@@ -43,12 +43,28 @@ pub async fn listen(session_id: String) -> Result<(), failure::Error> {
     // parse every incoming message and print them
     let print_ws_messages = {
         read.for_each(|message| async {
-            let message = message.unwrap().into_text().unwrap();
-            log::info!("{}", message);
-            let message: Result<protocol::Runtime, failure::Error> = serde_json::from_str(&message)
-                .map_err(|e| failure::format_err!("this event could not be parsed:\n{}", e));
-            if let Ok(protocol::Runtime::Event(event)) = message {
-                println!("{}", event);
+            match message {
+                Ok(message) => {
+                    let message_text = message.into_text().unwrap();
+                    log::info!("{}", message_text);
+                    let parsed_message: Result<protocol::Runtime, failure::Error> =
+                        serde_json::from_str(&message_text).map_err(|e| {
+                            failure::format_err!("this event could not be parsed:\n{}", e)
+                        });
+                    if let Ok(protocol::Runtime::Event(event)) = parsed_message {
+                        println!("{}", event);
+                    }
+                }
+                Err(error) => match error {
+                    tokio_tungstenite::tungstenite::Error::Protocol(msg) => {
+                        if msg == "Connection reset without closing handshake" {
+                            println!("please restart the connection");
```

Comment:
not sure what to do here... if it's reached this point, there's no going back because the connection has been reset (and i believe it's being reset by cloudflare, not us)

i want to start the `listen(session_id)` function over from the beginning at this point and maybe print a little "reconnecting" message with a spinner or something but i have _no_ idea how.